pkg/controller/ttlafterfinished/ttlafterfinished_controller.go

```diff
@@ -115,26 +116,30 @@ func (tc *Controller) Run(ctx context.Context, workers int) {
 	<-ctx.Done()
 }
 
-func (tc *Controller) addJob(obj interface{}) {
-	job := obj.(*batch.Job)
-	klog.V(4).Infof("Adding job %s/%s", job.Namespace, job.Name)
+func (tc *Controller) addJob(logger klog.Logger) func(obj interface{}) {
```

Comment:
I understand why you are making this a function generator (to keep `AddFunc:    tc.addJob(logger)` above simple), but it has the downsides that one cannot set breakpoints in `Controller.addJob` anymore and that this diff becomes larger than it has to be.

Let's instead use `func (tc *Controller) addJob(logger klog.Logger, obj interface{}) {` here and call it with:
```
AddFunc:    func(obj interface{} {
    tc.addJob(logger, obj)
},
```